src/map/utils/petutils.cpp

```diff
@@ -895,6 +897,43 @@ namespace petutils
         PPet->health.maxhp = (int16)(raceStat + jobStat + bonusStat + sJobStat);
         PPet->health.hp    = PPet->health.maxhp;
 
+        if (PPet->m_PetID == PETID_FIRESPIRIT || PPet->m_PetID == PETID_EARTHSPIRIT || PPet->m_PetID == PETID_AIRSPIRIT ||
```

Comment:
Are spirit IDs all side by side in the database? If so you can search if value is in between a range rather than checking on wvery value.